polyfill/lib/ecmascript.mjs

```diff
@@ -64,6 +67,232 @@ export const ES = ObjectAssign(ObjectAssign(ObjectAssign({}, Cast), ES2019), {
   IsDateTime: (item) => HasSlot(item, YEAR, MONTH, DAY, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND),
   IsYearMonth: (item) => HasSlot(item, YEAR, MONTH) && !HasSlot(item, DAY),
   IsMonthDay: (item) => HasSlot(item, MONTH, DAY) && !HasSlot(item, YEAR),
+  ToTimeZone: (item) => {
+    if (ES.IsTimeZone(item)) return item;
+    const isoString = ES.ToString(item);
+    const match = PARSE.timezone.exec(isoString);
+    if (!match) return new TemporalTimeZone(isoString);
+    if (!match) throw new RangeError(`invalid absolute: ${iso}`);
+    const zone = match[1] ? 'UTC' : match[3] || match[2];
```

Comment:
This can't work - the regex only has two capturing groups.

Reply:
Nope:

the **Z** is the first, the **+00:00** (offset) is the second and **[Continent/City]** (timezone) is the third. 

If the first exists, then 2 & 3 won't. If 3 exists use that otherwise use 2.

Reply:
The "Z" part is non-capturing:
```js
const zonesplit = /(?:Z|(?:([+-]\d{1,2}\:?\d{2})(?:\[([^\]\s]+)\])?))/;
```